src/condor_utils/proc_family_direct_cgroup_v1.cpp

```diff
@@ -0,0 +1,485 @@
+/***************************************************************
+ *
+ * Copyright (C) 1990-2022, Condor Team, Computer Sciences Department,
+ * University of Wisconsin-Madison, WI.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); you
+ * may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ * 
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ***************************************************************/
+
+
+#include "condor_common.h"
+#include "condor_daemon_core.h"
+#include "condor_uid.h"
+#include "directory.h"
+#include "proc_family_direct_cgroup_v1.h"
+#include <numeric>
+#include <array>
+
+#include <filesystem>
+
+namespace stdfs = std::filesystem;
+
+static std::map<pid_t, std::string> cgroup_map;
+
+static stdfs::path cgroup_mount_point() {
+	return "/sys/fs/cgroup";
+}
+
+// The controllers we use.  We manipulate and measure all
+// of these except for blkio.  We believe that merely 
+// putting the job into the blkio controller will give
+// job-level fair share to filesystem io.
+static std::array<const std::string, 4> controllers {
+		"memory", "cpu,cpuacct", "freezer", "blkio"};
+
+// Note that even root can't remove control group *files*,
+// only directories.
+void fullyRemoveCgroup(const stdfs::path &absCgroup) {
+
+	// Common case, it shouldn't still exist, we're done
+	if (!stdfs::exists(absCgroup)) {
+		return;
+	}
+
+	// Otherwise, we need to depth-firstly remove any subdirs
+	for (auto const& entry: std::filesystem::directory_iterator{absCgroup})
+    {
```

Comment:
This seems unlikely to be what you intended.
```suggestion
	for (auto const& entry: std::filesystem::directory_iterator{absCgroup}) {
```
?